R/bar_plot_emission_profile.R

```diff
@@ -0,0 +1,72 @@
+#' Create a horizontal stacked bar chart without financial data
+#'
+#' Generate a horizontal stacked bar chart showing the distribution of
+#' the emission risk profiles risks for one or several benchmarks.
+#'
+#' @param data A data frame like [without_financial].
+#' @param benchmarks A character vector specifying the benchmarks for which the
+#' emission profiles will be plotted. The user can choose from one to several
+#' benchmark(s) to be plotted.
+#'
+#' @return A [ggplot] object.
+#'
+#' @export
+#'
+#' @examples
+#' benchmarks <- c("all", "unit", "isic_sec")
+#' bar_plot_emission_profile(without_financial, benchmarks)
+bar_plot_emission_profile <- function(data,
+                          benchmarks = c(
+                            "all",
+                            "unit",
+                            "tilt_sec",
+                            "unit_tilt_sec",
+                            "isic_sec",
+                            "unit_isic_sec"
+                          )) {
+  benchmarks_arg <- arg_match(benchmarks, multiple = TRUE)
+  data <- data |>
+    na.omit()
+  warning("Rows with missing values in the data were dropped.")
```

Comment:
@lindadelacombaz note this warning throws even if the data has no missing value -- which is confusing. And if the data has missing values, then it's not clear which missing values were dropped.

The ggplot2 packages offfers good examples of when and how to throw a warning:

``` r
library(dplyr, warn.conflicts = FALSE)
library(ggplot2, warn.conflicts = FALSE)

# Informative warning
data <- tibble(x = c(1, NA))
ggplot(data) + geom_histogram(aes(x), bins = 30)
#> Warning: Removed 1 rows containing non-finite values (`stat_bin()`).
```

![](https://i.imgur.com/C8Rds3J.png)<!-- -->

``` r

# No warning
data <- tibble(x = c(1, 2))
ggplot(data) + geom_histogram(aes(x), bins = 30)
```

![](https://i.imgur.com/EPye6vr.png)<!-- -->

<sup>Created on 2024-02-13 with [reprex v2.1.0](https://reprex.tidyverse.org)</sup>

Once you arrive to a nice implementation you may extract a warning helper and reuse it in every function that needs to drop missing values. 

Or may be you can ignore the missing values and let ggplot2 handle them for you?

Reply:
Oh amazing. I will check this out!

Reply:
I have removed it since I have tried with geom_col() and if effectively takes care of the NA's as I wanted. Thanks!